<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<title>A2: Function glossary</title>

<script src="libs/jquery-1.11.0/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libs/bootstrap-3.3.1/css/readable.min.css" rel="stylesheet" />
<script src="libs/bootstrap-3.3.1/js/bootstrap.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/html5shiv.min.js"></script>
<script src="libs/bootstrap-3.3.1/shim/respond.min.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-52691601-2', 'auto');
  ga('send', 'pageview');

</script>

<!---

<style type="text/css">

/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
  padding-right: 50px;
  padding-left: -50px;
}
@media (max-width: 979px) {
  body {
    padding-top: 0;
  }
}

 offset scroll position for anchor links (for fixed navbar)  
@media (min-width: 980px) {
  .section h2 {
    padding-top: 52px;
    margin-top: -52px;
  }
  .section h3 {
    padding-top: 52px;
    margin-top: -52px;
  }
}


/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}

</style>
-->



<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 50px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h2 {
  padding-top: 55px;
  margin-top: -55px;
}
.section h3 {
  padding-top: 55px;
  margin-top: -55px;
}
/* don't use link color in navbar */
.dropdown-menu>li>a {
  color: black;
}
/* some padding for disqus */
#disqus_thread {
  margin-top: 45px;
}
</style>

<link rel="stylesheet" href="libs/font-awesome-4.1.0/css/font-awesome.min.css"/>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; background-color: #f8f8f8; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
pre, code { background-color: #f8f8f8; }
code > span.kw { color: #204a87; font-weight: bold; }
code > span.dt { color: #204a87; }
code > span.dv { color: #0000cf; }
code > span.bn { color: #0000cf; }
code > span.fl { color: #0000cf; }
code > span.ch { color: #4e9a06; }
code > span.st { color: #4e9a06; }
code > span.co { color: #8f5902; font-style: italic; }
code > span.ot { color: #8f5902; }
code > span.al { color: #ef2929; }
code > span.fu { color: #000000; }
code > span.er { font-weight: bold; }
</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>



</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img { 
  max-width:100%; 
  height: auto; 
}
</style>
<div class="container-fluid main-container">


<div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <!-- HOME -->
      <a class="navbar-brand" href="index.html"><b>Population Genetics in R</b></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <!-- Table of contents -->
        <li class="dropdown">
          <!---<a href="chapters" class="dropdown-toggle" data-toggle="dropdown">Chapters<b class="caret"></b></a>-->
          <a href="chapters" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Chapters <span class="caret"></span></a>
            <ol class="dropdown-menu">
               <li><a href="Preface.html">i. Preface</a></li>
               <li><a href="Introduction.html">1. Introduction</a></li>
               <li><a href="Getting_ready_to_use_R.html">2. Getting ready to use R</a></li>
               <li><a href="Data_Preparation.html">3. Data preparation</a></li>
               <li><a href="First_Steps.html">4. First steps</a></li>
               <li><a href="Population_Strata.html">5. Population strata and clone correction</a></li>
               <li><a href="Locus_Stats.html">6. Locus-based statistics and missing data</a></li>
               <li><a href="Genotypic_EvenRichDiv.html">7. Genotypic evenness, richness, and diversity</a></li>
               <li><a href="Linkage_disequilibrium.html">8. Linkage disequilibrium</a></li>
               <li><a href="Pop_Structure.html">9. Population structure</a></li>
               <li><a href="Minimum_Spanning_Networks.html">10. Minimum Spanning Networks</a></li>
               <li><a href="AMOVA.html">11. AMOVA</a></li>
               <li><a href="DAPC.html">12. Discriminant analysis of principal components (DAPC)</a></li>
            </ol>
        </li>
        <!-- Appendices dropdown -->
        <li class="dropdown">
          <!---<a href="appendices" class="dropdown-toggle" data-toggle="dropdown">Appendices<b class="caret"></b></a>-->
          <a href="appendices" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Appendices <span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
            <li><a href="Data_sets.html">A1. Data sets</a></li>
            <li><a href="funpendix.html">A2. Function glossary</a></li>
            <li><a href="Intro_to_R.html">A3. Introduction to R</a></li>
          </ul>
        </li>
        <!-- Author button -->
        <li class="dropdown">
        	<a href="about" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">About<span class="caret"></span></a>
          <ul class = "dropdown-menu", role = "menu">
        	<li><a href="Authors.html">Authors</a></li>
        	<li>
        		<a href="https://github.com/grunwaldlab/Population_Genetics_in_R", target = "blank">
		    	Source code
		    	</a>
		    </li>
		   </ul>
        </li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- 

old rmarkdown version: 

devtools::install_github("rstudio/rmarkdown", ref = "b96214b9ac86b437067a0aa21442203f52face83")


 -->
<!-- <div class="container">
<div class="row">
<div class="span10 offset1"> -->

<div id="header">
<h1 class="title">A2: Function glossary</h1>
</div>

<div id="TOC">
<ul>
<li><a href="#navigation-in-r">Navigation in R</a></li>
<li><a href="#poppr-functions"><em>Poppr</em> functions</a></li>
</ul>
</div>

<!---To quit R you can either use the Rstudio>Quit Rstudio pulldown menu command or execute âŒ˜+Q (OS X) or CTRL+Q (PC). 

Navigation in R
----

---->
<!--```{r, eval=FALSE}
getwd()        # this command will print the current working directory
ls()           # list objects in the current workspace
help(options)  # learn about available options
history()      # viewing your command history
data(*dsname*) # access builtin database *dsname*; must have pacakge loaded containing it
q()            # quit your R session

# default for saving history is ".Rhistory"
savehistory(file="myfile")  # save your command history for whole session
loadhistory(file= "myfile") # recall command history save

# Package specific information
installed.packages()[,c("Package","Version")] #list packages & versions installed
browseVignettes()                     # see vignettes from all packages

#Help on functions
help(functionname)     # Display documentation for a function
args(functionname)     # What are the arguments for a given function
example(functionname)  # see examples of function calls
```
-->
<div id="navigation-in-r" class="section level2">
<h2>Navigation in R</h2>
<p>Our primer is heavily based on the <em>poppr</em> and <em>adegenet</em> packages. To get help on any of their functions type a question mark before the empty function call as in:</p>
<pre class="sourceCode r"><code class="sourceCode r">?mlg <span class="co"># open the R documentiaon of the function mlg</span></code></pre>
<p>Other information can be obtained with the following commands.</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseVignettes</span>(<span class="dt">package =</span> <span class="st">&#39;poppr&#39;</span>) <span class="co"># see vignettes for poppr</span>
<span class="kw">vignette</span>(<span class="st">&#39;poppr_manual&#39;</span>)           <span class="co"># see manual for poppr</span>
<span class="kw">help</span>(<span class="dt">package=</span><span class="st">&quot;poppr&quot;</span>)              <span class="co"># shows all functions contained in poppr package</span></code></pre>
</div>
<div id="poppr-functions" class="section level2">
<h2><em>Poppr</em> functions</h2>
<div id="importexport" class="section level3">
<h3>Import/Export</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>getfile</code></td>
<td align="left">Provides a quick GUI to grab files for import</td>
</tr>
<tr class="even">
<td align="left"><code>read.genalex</code></td>
<td align="left">Read <em>GenAlEx</em> formatted csv files to a genind object</td>
</tr>
<tr class="odd">
<td align="left"><code>genind2genalex</code></td>
<td align="left">Converts genind objects to <em>GenAlEx</em> formatted csv files</td>
</tr>
</tbody>
</table>
</div>
<div id="manipulation" class="section level3">
<h3>Manipulation</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>as.genclone</code></td>
<td align="left">Converts genind objects to genclone objects</td>
</tr>
<tr class="even">
<td align="left"><code>setPop</code></td>
<td align="left">Set the population using defined strata</td>
</tr>
<tr class="odd">
<td align="left"><code>splitStrata</code></td>
<td align="left">Split concatenated strata imported as a population</td>
</tr>
<tr class="even">
<td align="left"><code>strata</code></td>
<td align="left">Define or extract population strata of a genind object</td>
</tr>
<tr class="odd">
<td align="left"><code>addStrata</code></td>
<td align="left">Add a vector or data frame to an existing strata</td>
</tr>
<tr class="even">
<td align="left"><code>nameStrata</code></td>
<td align="left">Rename a population strata</td>
</tr>
<tr class="odd">
<td align="left"><code>missingno</code></td>
<td align="left">Handles missing data</td>
</tr>
<tr class="even">
<td align="left"><code>clonecorrect</code></td>
<td align="left">Clone censor at a specified population stratum</td>
</tr>
<tr class="odd">
<td align="left"><code>informloci</code></td>
<td align="left">Detects and removes phylogenetically uninformative loci</td>
</tr>
<tr class="even">
<td align="left"><code>popsub</code></td>
<td align="left">Subsets genind objects by population</td>
</tr>
<tr class="odd">
<td align="left"><code>shufflepop</code></td>
<td align="left">Shuffles genotypes at each locus using four different shuffling algorithms</td>
</tr>
<tr class="even">
<td align="left"><code>recode_polyploids</code></td>
<td align="left">Recodes an imported polyploid data set to observed allelic frequencies</td>
</tr>
</tbody>
</table>
</div>
<div id="distances" class="section level3">
<h3>Distances</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>bruvo.dist</code></td>
<td align="left">Bruvoâ€™s distance</td>
</tr>
<tr class="even">
<td align="left"><code>diss.dist</code></td>
<td align="left">Absolute genetic distance (see <code>provesti.dist</code>)</td>
</tr>
<tr class="odd">
<td align="left"><code>nei.dist</code></td>
<td align="left">Neiâ€™s 1978 genetic distance</td>
</tr>
<tr class="even">
<td align="left"><code>rogers.dist</code></td>
<td align="left">Rogersâ€™ euclidean distance</td>
</tr>
<tr class="odd">
<td align="left"><code>reynolds.dist</code></td>
<td align="left">Reynoldsâ€™ coancestry distance</td>
</tr>
<tr class="even">
<td align="left"><code>edwards.dist</code></td>
<td align="left">Edwardsâ€™ angular distance</td>
</tr>
<tr class="odd">
<td align="left"><code>provesti.dist</code></td>
<td align="left">Provestiâ€™s absolute genetic distance</td>
</tr>
</tbody>
</table>
</div>
<div id="bootstrapping" class="section level3">
<h3>Bootstrapping</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>aboot</code></td>
<td align="left">Creates a bootstrapped dendrogram for any distance measure</td>
</tr>
<tr class="even">
<td align="left"><code>bruvo.boot</code></td>
<td align="left">Produces dendrograms with bootstrap support based on Bruvoâ€™s distance</td>
</tr>
</tbody>
</table>
</div>
<div id="analysis" class="section level3">
<h3>Analysis</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>poppr.amova</code></td>
<td align="left">Analysis of Molecular Variance (as implemented in ade4)</td>
</tr>
<tr class="even">
<td align="left"><code>ia</code></td>
<td align="left">Calculates the index of association</td>
</tr>
<tr class="odd">
<td align="left"><code>mlg</code></td>
<td align="left">Calculates the number of multilocus genotypes</td>
</tr>
<tr class="even">
<td align="left"><code>mlg.crosspop</code></td>
<td align="left">Finds all multilocus genotypes that cross populations</td>
</tr>
<tr class="odd">
<td align="left"><code>mlg.table</code></td>
<td align="left">Returns a table of populations by multilocus genotypes</td>
</tr>
<tr class="even">
<td align="left"><code>mlg.vector</code></td>
<td align="left">Returns a vector of a numeric multilocus genotype assignment for each individual</td>
</tr>
<tr class="odd">
<td align="left"><code>mlg.id</code></td>
<td align="left">Finds all individuals associated with each multilocus genotype</td>
</tr>
<tr class="even">
<td align="left"><code>poppr</code></td>
<td align="left">Returns a diversity table by population</td>
</tr>
<tr class="odd">
<td align="left"><code>poppr.all</code></td>
<td align="left">Returns a diversity table by population for all compatible files specified</td>
</tr>
<tr class="even">
<td align="left"><code>private_alleles</code></td>
<td align="left">Tabulates the occurences of alleles that only occur in one population</td>
</tr>
<tr class="odd">
<td align="left"><code>locus_table</code></td>
<td align="left">Create a table of summary statistics per locus</td>
</tr>
</tbody>
</table>
</div>
<div id="visualization" class="section level3">
<h3>Visualization</h3>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>plot_poppr_msn</code></td>
<td align="left">Plots minimum spanning networks produced in poppr with scale bar and legend</td>
</tr>
<tr class="even">
<td align="left"><code>greycurve</code></td>
<td align="left">Helper to determine the appropriate parameters for adjusting the grey level for msn functions</td>
</tr>
<tr class="odd">
<td align="left"><code>bruvo.msn</code></td>
<td align="left">Produces minimum spanning networks based off Bruvoâ€™s distance colored by population</td>
</tr>
<tr class="even">
<td align="left"><code>poppr.msn</code></td>
<td align="left">Produces a minimum spanning network for any pairwise distance matrix related to the data</td>
</tr>
<tr class="odd">
<td align="left"><code>info_table</code></td>
<td align="left">Creates a heatmap representing missing data or observed ploidy</td>
</tr>
<tr class="even">
<td align="left"><code>genotype_curve</code></td>
<td align="left">Creates a series of boxplots to demonstrate how many markers are needed to represent the diversity of your data</td>
</tr>
</tbody>
</table>
<div class="references">

</div>
</div>
</div>

<!-- </div>
</div>
</div>
 -->

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});

</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
